src/core/meshcop/dataset.cpp

```diff
@@ -66,6 +66,7 @@ Error Dataset::Info::GenerateRandom(Instance &aInstance)
     mActiveTimestamp.mAuthoritative = false;
     mChannel                        = preferredChannels.ChooseRandomChannel();
     mChannelMask                    = supportedChannels.GetMask();
+    mWakeupChannel                  = OPENTHREAD_CONFIG_DEFAULT_WAKEUP_CHANNEL;
```

Comment:
Should we pick randomly here rather than use default? (same way we pick `mChannel` from `preferredChannels.ChooseRandomChannel()`)?

Reply:
I don't think it's the same situation since the spec doesn't mandate a default PAN channel, while a default wake-up channel is specified.

Reply:
Fair point. In the prototype we initially made sure the wake-up channel was two channels apart from the PAN channel (to minimize side-band interference): https://github.com/nrfconnect/sdk-openthread/blob/collab-thread-s2s/src/core/mac/channel_mask.cpp#L103 It was then set to a fixed value to avoid corner cases of devices not parsing the new dataset.

But this is a grey zone when comes to the specification, I guess it can be considered implementation specific decision. How about just reusing `GetNextChannelWithRollOver(GetNextChannelWithRollOver(aChannel))` from the prototype? Or just `ChooseRandomChannel(void)` but iterating until it's at least two channels apart from PAN channel?

Reply:
Okay, for now I'm just setting a random channel. Applications wishing something more specific can always alter the dataset.